backend/backend.go

```diff
@@ -930,3 +933,75 @@ func (backend *Backend) SetWatchonly(rootFingerprint []byte, watchonly bool) err
 		&t,
 	)
 }
+
+// requestAddressScriptTypeMap maps from format codes specified by request-address js library
+// to our own script type codes. See https://www.npmjs.com/package/request-address
+var requestAddressScriptTypeMap = map[string]signing.ScriptType{
+	"p2pkh":  signing.ScriptTypeP2PKH,
+	"p2wpkh": signing.ScriptTypeP2WPKH,
+	"p2sh":   signing.ScriptTypeP2WPKHP2SH,
+	"p2tr":   signing.ScriptTypeP2TR,
+}
+
+// defaultScriptType is the default type used for address sharing and verification, if the
+// `format` param is not defined.
+var defaultScriptType = "p2wpkh"
+
+// SignBTCAddress returns an unused address and makes the user sign a message to prove ownership.
+// Input params:
+//
+//	`account` is the account from which the address is derived, and that will be linked to the Pocket order.
+//	`message` is the message that will be signed by the user with the private key linked to the address.
+//	`format` is the script type that should be used in the address derivation, as received by the widget
+//		(see https://github.com/pocketbitcoin/request-address#requestaddressv0messagescripttype).
+//		If format is empty, native segwit type is used as a fallback.
+//
+// Returned values:
+//
+//	#1: is the first unused address corresponding to the account and the script type identified by the input values.
+//	#2: base64 encoding of the message signature, obtained using the private key linked to the address.
+//	#3: is an optional error that could be generated during the execution of the function.
+func (backend *Backend) SignBTCAddress(account accounts.Interface, message string, format string) (string, string, error) {
```

Comment:
`backend` is not used at all, so this does not need to be a `Backend` method.

Reply:
Uh, right! Changed and moved to btc/account.go as suggested in the other comment